Sources/OpenCombine/Publishers/Future.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  Future.swift
+//  
+//
+//  Created by Max Desiatov on 24/11/2019.
+//
+
+import COpenCombineHelpers
+
+/// A publisher that eventually produces one value and then finishes or fails.
+public final class Future<Output, Failure>: Publisher where Failure: Error {
+
+    public typealias Promise = (Result<Output, Failure>) -> Void
+
+    private let _lock = UnfairRecursiveLock.allocate()
```

Comment:
You never deallocate this lock. It must be deallocated manually. Can you add a `deinit`?

```swift
deinit {
    _lock.deallocate()
}
```